crypto/rand/rand_lib.c

```diff
@@ -49,13 +71,18 @@ const RAND_METHOD *RAND_get_rand_method(void)
 #endif
             default_RAND_meth = RAND_OpenSSL();
     }
+    CRYPTO_THREAD_unlock(rand_meth_lock);
     return default_RAND_meth;
```

Comment:
Are assignments atomic? This read of the variable is unprotected by a lock, so I wonder what happens if RAND_set_rand_method() is executed at the same time as this read is taking place.